tuftool/src/add_role.rs

```diff
@@ -0,0 +1,258 @@
+    where
+        T: Transport,
+    {
+        let paths = if let Some(paths) = &self.paths {
+            PathSet::Paths(paths.clone())
+        } else if let Some(path_hash_prefixes) = &self.path_hash_prefixes {
+            PathSet::PathHashPrefixes(path_hash_prefixes.clone())
+        } else {
+            // Should warn that no paths are being delegated
+            PathSet::Paths(Vec::new())
+        };
+        let updated_role = editor
+            .add_role(
+                &self.delegatee,
+                self.indir.as_str(),
+                paths,
+                self.threshold,
+                None,
+            )
+            .context(error::LoadMetadata)?
+            .version(self.version)
+            .expires(self.expires)
+            .sign(&self.keys)
+            .context(error::SignRepo)?;
+        let metadata_destination_out = &self.outdir.join("metadata");
+        updated_role
+            .write(metadata_destination_out, false)
+            .context(error::WriteRoles {
+                roles: [self.delegatee.clone(), role.to_string()].to_vec(),
+            })?;
+
+        Ok(())
+    }
+
+    /// Adds a role to metadata using repo Editor
+    fn with_repo_editor<T>(&self, role: &str, mut editor: RepositoryEditor<'_, T>) -> Result<()>
+    where
+        T: Transport,
+    {
+        // Since we are using repo editor we will sign snapshot and timestamp
+        // Check to make sure all versions and expirations are present
+        let snapshot_version = self.snapshot_version.context(error::Missing {
+            what: "snapshot version".to_string(),
+        })?;
+        let snapshot_expires = self.snapshot_expires.context(error::Missing {
+            what: "snapshot expires".to_string(),
+        })?;
+        let timestamp_version = self.timestamp_version.context(error::Missing {
+            what: "timestamp version".to_string(),
+        })?;
+        let timestamp_expires = self.timestamp_expires.context(error::Missing {
+            what: "timestamp expires".to_string(),
+        })?;
+        let paths = if let Some(paths) = &self.paths {
+            PathSet::Paths(paths.clone())
+        } else if let Some(path_hash_prefixes) = &self.path_hash_prefixes {
+            PathSet::PathHashPrefixes(path_hash_prefixes.clone())
+        } else {
+            // Should warn that no paths are being delegated
+            PathSet::Paths(Vec::new())
+        };
+        // change editor's targets to role
+        editor
+            .targets_version(self.version)
+            .context(error::DelegationStructure)?
+            .targets_expires(self.expires)
+            .context(error::DelegationStructure)?
+            .sign_targets_editor(&self.keys)
+            .context(error::DelegateeNotFound {
+                role: role.to_string(),
+            })?
+            .change_delegated_targets(role)
+            .context(error::DelegateeNotFound {
+                role: role.to_string(),
```

Comment:
I know it's slightly longer, but could we split this call into 2 steps with comments to make it clear what's happening?

1) Update `Targets` version, expiration, sign
2) Change targets role to delegated targets